patches/server/0738-Add-configurable-height-for-slime-spawn.patch

```diff
@@ -0,0 +1,60 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Doc <nachito94@msn.com>
+Date: Mon, 2 Aug 2021 11:24:39 -0400
+Subject: [PATCH] Add configurable height for slime spawn
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index c334f29c69c1e6e3fe55cd6695e7df400cf36058..ab50da2a07dff960a73d0fb47849624f302cb09c 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -345,6 +345,16 @@ public class PaperWorldConfig {
+         allChunksAreSlimeChunks = getBoolean("all-chunks-are-slime-chunks", false);
+     }
+ 
++    public double slimeMaxSpawnHeightInSwamp;
++    public double slimeMinSpawnHeightInSwamp;
++    public double slimeMaxSpawnHeightInSlimeChunks;
++    private void slimeSpawnHeight() {
++        slimeMaxSpawnHeightInSwamp = getDouble("slime-spawn-height.swamp-biome.maximum", 70);
++        slimeMinSpawnHeightInSwamp = getDouble("slime-spawn-height.swamp-biome.minimum", 50);
++        slimeMaxSpawnHeightInSlimeChunks = getDouble("slime-spawn-height.slime-chunk.maximum", 40);
++    }
++
++
+     public int portalSearchRadius;
+     public int portalCreateRadius;
+     public boolean portalSearchVanillaDimensionScaling;
+diff --git a/src/main/java/net/minecraft/world/entity/monster/Slime.java b/src/main/java/net/minecraft/world/entity/monster/Slime.java
+index e1a593b464c35f68b22e84a09f99ee72af73da32..9c95c5e8550b48baad84be9e97be7ee169a70fb3 100644
+--- a/src/main/java/net/minecraft/world/entity/monster/Slime.java
++++ b/src/main/java/net/minecraft/world/entity/monster/Slime.java
+@@ -326,9 +326,13 @@ public class Slime extends Mob implements Enemy {
+ 
+     public static boolean checkSlimeSpawnRules(EntityType<Slime> type, LevelAccessor world, MobSpawnType spawnReason, BlockPos pos, Random random) {
+         if (world.getDifficulty() != Difficulty.PEACEFUL) {
+-            if (Objects.equals(world.getBiomeName(pos), Optional.of(Biomes.SWAMP)) && pos.getY() > 50 && pos.getY() < 70 && random.nextFloat() < 0.5F && random.nextFloat() < world.getMoonBrightness() && world.getMaxLocalRawBrightness(pos) <= random.nextInt(8)) {
++            // Paper start - Replace rules for Height in Swamp Biome
++            final double maxHeightSwamp = world.getMinecraftWorld().paperConfig.slimeMaxSpawnHeightInSwamp;
++            final double minHeightSwamp = world.getMinecraftWorld().paperConfig.slimeMinSpawnHeightInSwamp;
++            if (Objects.equals(world.getBiomeName(pos), Optional.of(Biomes.SWAMP)) && pos.getY() > minHeightSwamp && pos.getY() < maxHeightSwamp && random.nextFloat() < 0.5F && random.nextFloat() < world.getMoonBrightness() && world.getMaxLocalRawBrightness(pos) <= random.nextInt(8)) {
+                 return checkMobSpawnRules(type, world, spawnReason, pos, random);
+             }
++            // Paper end
```

Comment:
Your diff ends earlier, including the return statement and closing brackets in your paper comment block just enlarges the diff. You should probably end the diff directly after the if.

I don't know how feasible it is to one-line these changes, as the if statment was already quite long to begin with, I'll have to check once I have access to a wide enough screen.

Reply:
(on a not-totally-unrelated note, we should add a method for getting the PaperConfig to LevelAccessor)

Reply:
Ok i try make this changes... i dont fully understand CONTRIBUTING about change patchs (in this case only get errors)

i think is ready.